### PR TITLE
Copy LruMap from scala-maxmind-iplookups

### PR DESCRIPTION
**Pull request by three**

### CODE REVIEW

File: build.sbt
Comment (BenFradet) [inline build.sbt]: 2.12.6

File: src/main/scala/com.snowplowanalytics.maxmind.lrumap/LruMap.scala
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics.maxmind.lrumap/LruMap.scala]: I don't think this is needed either?

File: README.md
Comment (BenFradet) [inline README.md]: this is just memory allocation, I don't think we need to consider that a side-effect

File: src/main/scala/com.snowplowanalytics.maxmind.lrumap/LruMap.scala
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics.maxmind.lrumap/LruMap.scala]: can you add some scaladoc to the object / class and functions please?

File: src/main/scala/com.snowplowanalytics.maxmind.lrumap/LruMap.scala
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics.maxmind.lrumap/LruMap.scala]: why is `maxSize` needed? also no need to have `val`s

**Review by BenFradet — APPROVED** [review]
mostly nits and docs :+1:

**Comment by BenFradet** [comment]
LGTM, I'll let @chuwy have a second look

File: LICENSE-2.0.txt
Comment (chuwy) [inline LICENSE-2.0.txt]: 2012-2018?

File: README.md
Comment (chuwy) [inline README.md]: It is a very common convention to append `()` to `Unit`-side-effecting method invocation: `effect.unsafeRunSync()`

**Review by chuwy — APPROVED** [review]
Looks perfect.

**Comment by BenFradet** [comment]
Feel free to squash everything and tag a 0.1.0-M1

**Comment by BenFradet** [comment]
Please don't merge yourself there was still a lot that needed doing before going to master

**Comment by three** [comment]
Sorry! I thought you meant you wanted me to squash into master. Should I revert master back to 177f58f?

**Comment by BenFradet** [comment]
no I just meant squash here, no need to revert, please add a CHANGELOG and rename the commit to "Initial commit".

**Comment by BenFradet** [comment]
Also we usually don't merge before we have the go ahead to publish a final version e.g. not an release candidate or milestone version.

**Comment by three** [comment]
I squashed everything to one commit with the changelog and included it in the initial branch. Is there anything else I should do to get the final version ready?

**Comment by BenFradet** [comment]
This should be on master since the initial branch has already been merged...

**Comment by three** [comment]
Okay, I pushed the commit to master.

**Comment by BenFradet** [comment]
heads up that the changelog should look like

```
Version 0.1.0 (2017-08-31)
--------------------------
Initial release
```

but we need to wait on #2 and #3 to release a 0.1.0-M1 and then a final version.

**Comment by three** [comment]
I edited the changelog and put the squashed commit on master.

**Pull request merged**
